File: Python/Sliding Window/Count Subarrays Where Max Element Appears at Least K Times.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        maxi = max(nums)
        left =0
        ans =0
        maxElemInWin = 0
        for right in range(0,len(nums)):
            if nums[right] ==maxi:
                maxElemInWin+=1
            while maxElemInWin == k:
                if nums[left] ==maxi:
                    maxElemInWin-=1
                left+=1
            ans+=left
        return ans

        

File: CPP/Sliding Window/Count Subarrays Where Max Element Appears at Least K Times.cpp
class Solution {
public:
    long long countSubarrays(vector<int>& nums, int k) {
        int maxi = *max_element(nums.begin(),nums.end());
        int left=0; int maxwindowEle =0;
        long long ans =0;
        for(int right=0;right<nums.size();right++){
            if(nums[right]==maxi)maxwindowEle++;
            while(maxwindowEle==k){
                if(nums[left]==maxi)maxwindowEle--;
                left++;
            }
            ans+=left;
        }
        return ans;
    }
}
